### DISCUSSION TITLE
expandRecord usage

### DISCUSSION THREAD

**Discussion by gregoryforel** [pocketbase/pocketbase#4215] [Q&A] [answered]
I would like to use Pocketbase as a framework. Originally, I had this query using the pocketbase js sdk, it works fine:
```typescript
const learningPathResult = await pb.collection('learning_path').getOne<LearningPathResponse>(learningPathId, {
    expand: 'learning_path_lesson(learning_path).lesson,learning_path_lesson(learning_path).chapter'
})
```

How can I translate it into a new route? I am stuck with the expandRecord function. Here's what I have:
```javascript
routerAdd("GET", "/get_learning_path_hierarchy/:id", (c) => {
    let id = c.pathParam("id")
   // this works:
    const learningPath = $app.dao().findRecordById("learning_path", id)


    //  not sure how to translate my learning_path_lesson(learning_path).lesson,learning_path_lesson(learning_path).chapter expand here:
    $app.dao().expandRecord(learningPath, ["learning_path_lesson(learning_path).lesson", "learning_path_lesson(learning_path"], null)
   
    return c.json(200, { "message": learningPath })
})
```

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear from your example what actually doesn't work and what you are trying to do in general.

Also note that calling manually `$app.dao().expandRecord` will expand the specified relations data into the record without checking the request context or the related collection View API rule.

If you want to have the same behavior as the client-side request but with just predefined expand list, you can try the [`$apis.enrichRecord` helper](https://pocketbase.io/docs/js-routing/#enrich-records).

To check for a more detailed errors, you can also try starting the executable with `--dev`.

**Reply by gregoryforel** [reply]
Thank you, I had not realized there was the enrichRecord method, it worked perfectly.
